background/get-userscripts.js

```diff
@@ -40,13 +41,7 @@ function userscriptMatches(data, matches) {
 }
 
 function urlMatchesPattern(_pattern, _url) {
-  const pattern = _pattern.split("/");
-  const url = _url.split("/");
-  while (pattern.length) {
-    const p = pattern.shift();
-    const q = url.shift();
-    if (p !== q && p !== "*") return false;
-  }
-  return true;
-}
-window.urlMatchesPattern = urlMatchesPattern;
+  _url = ((_pattern.endsWith("/")) && (!_url.endsWith("/"))) ? `${_url}/` : _url;
+  _url = _url.split("#")[0].split("?")[0];
```

Comment:
Seems pretty weird that the library wouldn't handle hashes...?
If the library is kinda "dumb" like this one, wouldn't we better off using the built-in URL constructor, add an implicit "/" at the end of the path, and compare the paths with the previous code?

Reply:
Okay. So, can you give some example code?

Reply:
See https://developer.mozilla.org/en-US/docs/Web/API/URL
Try getting the pathname for the URLs, add an "/" to the URL if it doesn't have one, then compare them with the previous code I written (or write your own, if you'd like)